### DISCUSSION TITLE
LambdaStartupAttribute support for Configure method.

### DISCUSSION THREAD

**Discussion by manuquintero** [aws/aws-lambda-dotnet#1688] [Q&A] [answered]
Hi, 
In a Amazon.Lambda.Annotations application, I am using LambdaStartup but it looks like it only supports ConfigureServices, how can I use Configure so I can configure other items like UseAuthentication() and UseAuthorization() 

Thanks

**Comment by ashishdhingra** [comment, marked as answer]
@manuquintero Good morning. Per my analysis and available documentation, 
- Lambda annotations relies on source generators to translate your annotated Lambda functions to the regular Lambda programming model (refer https://aws.amazon.com/blogs/developer/net-lambda-annotations-framework/).
- Decorating a class with `Amazon.Lambda.Annotations.LambdaStartup` attribute causes source generation process to invoke <<class>>.ConfigureServices() in the generated code for LambdaFunctions. Below is an example:
```C#
...
namespace SomeNamespace
{
    public class Functions_Add_Generated
    {
        private readonly ServiceProvider serviceProvider;

        public Functions_Add_Generated()
        {
            SetExecutionEnvironment();
            var services = new ServiceCollection();

            ...
            var startup = new LambdaAnnotations_Issue1368.Startup();
            startup.ConfigureServices(services);
            serviceProvider = services.BuildServiceProvider();
        }

        public Amazon.Lambda.APIGatewayEvents.APIGatewayHttpApiV2ProxyResponse Add(Amazon.Lambda.APIGatewayEvents.APIGatewayHttpApiV2ProxyRequest __request__, Amazon.Lambda.Core.ILambdaContext __context__)
        {
...
    }
}
```
- Source generation process also generates `serverless.template` to create API Gateway resources. So it's mapping HTTP API endpoints to generated Lambda functions.

Based on above understanding, I'm unsure if we could configure HTTP request pipeline to invoke `UseAuthentication()`, `UseAuthorization()`, etc. since it doesn't leverage the ASP.NET request pipeline.

Reference example (using both programming models): https://github.com/awsdocs/aws-doc-sdk-examples/tree/main/dotnetv3/cross-service/PhotoAssetManager

CCing @normj for validating the findings and any further inputs.

Thanks,
Ashish
